sw/device/lib/crypto/drivers/kmac_test.c

```diff
@@ -61,5 +63,46 @@ bool test_main(void) {
                     current_test_vector->digest_len);
   }
 
+  // Below we repeat some of the test above, but through higher-level
+  // cryptolib API. The gist is to test if glueing is done correctly.
+  for (i = 0; i < nist_kmac_nr_of_vectors; i++) {
+    current_test_vector = nist_kmac_vectors[i];
+
+    crypto_const_uint8_buf_t input_buf;
+    input_buf.data = current_test_vector->input_str;
+    input_buf.len = current_test_vector->input_str_len;
```

Comment:
nit: do we need these assignments? We could define an inner struct in `current_test_vector` with the same type as `input_buf` and just pass it to the function we want to call.

Reply:
I agree, but let me delay this effort to future PRs (updated progress-tracking issue #16410). The files on the test side (mainly .h and .c of `kmac_test` and `kmac_test_vectors`) are far from being stable and I change them quite frequently. Once I finish gluing KMAC driver to cryptolib API, I will also work on bringing more test vectors and I can focus on reusing the lib structs for the test vectors.